plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp

```diff
@@ -177,13 +178,30 @@ void DialogSelectTopics::on_change_domain_button_clicked()
 {
     DEBUG("Calling on_change_domain_button_clicked");
 
-    // Change domain connected
-    domain_id_selected_ =
-            static_cast<unsigned int>(ui->domainid_spin->value());
+    // Start a new dialog to select a domain
+    DialogSelectDomain* change_domain_dialog = new DialogSelectDomain(domain_id_connected_, this);
```

Comment:
please use safer RAII semantics for safety
either using `unique_ptr`:
`auto change_domain_dialog = std::make_unique<DialogSelectDomain>(domain_id_connected_, this);`

or plain object if DialogSelectDomain is not polymorphic:
`DialogSelectDomain change_domain_dialog(domain_id_connected_, this);`

Reply:
This follows Qt guides. I rather leave it this way.